File: runtime/scala/src/main/scala/dev/argon/verilization/scala_runtime/FormatReader.scala
package dev.argon.verilization.scala_runtime

import zio.{ZIO, Chunk}
import scala.{Int => SInt}

trait FormatReader[-R, +E] {
    def readByte(): ZIO[R, E, Byte]
    def readShort(): ZIO[R, E, Short]
    def readInt(): ZIO[R, E, SInt]
    def readLong(): ZIO[R, E, Long]
    def readBytes(count: SInt): ZIO[R, E, Chunk[Byte]]
}

File: runtime/java/src/main/java/dev/argon/verilization/java_runtime/FormatReader.java
package dev.argon.verilization.java_runtime;

import java.io.IOException;

public interface FormatReader {
	byte readByte() throws IOException;
	short readShort() throws IOException;
	int readInt() throws IOException;
	long readLong() throws IOException;
	byte[] readBytes(int count) throws IOException;
}
